vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp

```diff
@@ -411,29 +411,28 @@ bool TouchEventHandler::TagFromOriginalSource(
   winrt::IPropertyValue tag(nullptr);
 
   while (sourceElement) {
-    if (auto fe = sourceElement.try_as<xaml::FrameworkElement>()) {
-      tag = fe.GetValue(xaml::FrameworkElement::TagProperty()).try_as<winrt::IPropertyValue>();
-      if (tag) {
```

Comment:
I think the fix you want instead of reverting it all is to close the "if (try as framework element)" block right after assigning tag. That way if we didn't assign it, tag will still be null

Reply:
@asklar I think that will still skip all the `Inline` elements that I'm iterating over, and there will be no recursion.

Instead, we'd need a way of checking if the `TagProperty` is present on the current element, rather than check if the element is a `FrameworkElement` - we call `SetValue(xaml::FrameworkElement::TagProperty(), ..)` on all ShadowNodes, not just those whose XamlView can be cast to a `FrameworkElement` (here, specifically the `Span` elements).

I was thinking of checking this PR in, and immediately following up with a fix for the scenario in question, just for the sake of clarity/separation between PRs.

Do you know of a way to do such a check? Would `ReadLocalValue` be one? 
Side question, is it correct that we assign the `FE::TagProperty` to `Inline`s?

Reply:
in general it's not super kosher for a framework to take up a xaml property like Tag. We have a workitem to remove this: #4187

Reply:
Agreed, we should look into registering our own custom DP. For now, using ReadLocalValue to prevent the crash